src/csharp/Microsoft.Spark/Interop/Ipc/JsonSerDe.cs

```diff
@@ -2,70 +2,139 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Linq;
-using Newtonsoft.Json.Linq;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
 
 namespace Microsoft.Spark.Interop.Ipc
 {
     /// <summary>
-    /// Json.NET Serialization/Deserialization helper class.
+    /// System.Text.Json Serialization/Deserialization helper class.
     /// </summary>
     internal static class JsonSerDe
     {
+
         /// Note: Scala side uses JSortedObject when parsing JSON, so the properties
-        /// in JObject need to be sorted.
+        /// in JsonElement need to be sorted.
         /// <summary>
         /// Extension method to sort items in a JSON object by keys.
         /// </summary>
-        /// <param name="jObject"></param>
+        /// <param name="jsonElement"></param>
+        /// <returns></returns>
+        public static string SortProperties(this JsonElement jsonElement)
+        {
+            var output = new ArrayBufferWriter<byte>();
+            using (var writer = new Utf8JsonWriter(output))
+                jsonElement.SortPropertiesCore(writer);
+            return Encoding.UTF8.GetString(output.WrittenSpan.ToArray());
+        }
+
+        /// <summary>
+        /// Helper method to Parse a string into a JsonElement.
+        /// </summary>
+        /// <param name="json"></param>
         /// <returns></returns>
-        public static JObject SortProperties(this JObject jObject)
+        public static JsonElement Parse(string json)
+        {
+            using (var document = JsonDocument.Parse(json))
+                return document.RootElement.Clone();
+        }           
+
+        /// <summary>
+        /// Helper method to Parse an object into a JsonElement.
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
+        public static JsonElement Parse(object obj)
+        {
+            return Parse(JsonSerializer.Serialize(obj));
```

Comment:
Is there a better way than this? Seems quite inefficient. I thought of passing a `Dictionary<string, (JsonValueKind, object)>` in and then building the JsonElement with `Utf8JsonWriter` and writing the appropriate type but it seems a little verbose. Any ideas?

Reply:
Is this `Parse` method necessary (i.e. one that takes in an `object`)? Can all the call-sites instead write to a writer rather than creating an object?

If that is too inconvenient, one way to make things faster is to have a utf-8 based implementation (which avoids the unnecessary transcoding cost from utf-16 string to utf-8 on both sides (the serialize and the parse):
```C#
return Parse(JsonSerializer.SerializeToUtf8Bytes(obj));

private static JsonElement Parse(byte[] utf8Json)
{
   using (var document = JsonDocument.Parse(utf8Json))
      return document.RootElement.Clone();
}
```